node/node.go

```diff
@@ -115,6 +115,9 @@ type AlgorandFullNode struct {
 
 	log logging.Logger
 
+	// use syncMu for locking lastRoundTimestamp and hasSyncedSinceStartup
+	// syncMu added so OnNewBlock wouldn't be blocked by oldKeyDeletionThread during catchup
+	syncMu                deadlock.Mutex
```

Comment:
could you change the variable name to something more concrete. ( since all mutexes are meant for syncing.. )
maybe lastRoundTimeMu ?

Reply:
the two variables together might be 'sync status', so this could be the 'sync status mutex'